mesonbuild/compilers/c.py

```diff
@@ -1016,7 +1018,7 @@ def find_library_real(self, libname, env, extra_dirs, code, libtype):
         if not extra_dirs or libname in self.internal_libs:
             args = ['-l' + libname]
             largs = self.linker_to_compiler_args(self.get_allow_undefined_link_args())
-            if self.links(code, env, extra_args=(args + largs)):
+            if self.links(code, env, extra_args=(args + largs))[0]:
```

Comment:
I think this needs to by-pass the persistent cache, because find_library may only cache positive results.

Reply:
Done